Failing test(s): TestAccCloudRunService_cloudRunServiceGpuExample

### Impacted tests

TestAccCloudRunService_cloudRunServiceGpuExample


### Affected Resource(s)

google_cloud_run_service

### Failure rates

- GA: 
- Beta: 100%


### Message(s)

```
=== RUN   TestAccCloudRunService_cloudRunServiceGpuExample
=== PAUSE TestAccCloudRunService_cloudRunServiceGpuExample
=== CONT  TestAccCloudRunService_cloudRunServiceGpuExample
    resource_cloud_run_service_generated_test.go:90: Step 1/2 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_cloud_run_service.default will be updated in-place
          ~ resource "google_cloud_run_service" "default" {
                id                         = "locations/us-central1/namespaces/ci-test-project-nightly-beta/services/tf-test-cloudrun-srvowr0f1fhjp"
                name                       = "tf-test-cloudrun-srvowr0f1fhjp"
                # (4 unchanged attributes hidden)
        
              ~ metadata {
                  + labels                = {}
                    # (9 unchanged attributes hidden)
                }
        
              ~ template {
                  ~ metadata {
                      ~ annotations      = {
                          - "run.googleapis.com/gpu-zonal-redundancy-disabled" = "true" -> null
                            # (2 unchanged elements hidden)
                        }
                        name             = null
                        # (6 unchanged attributes hidden)
                    }
                  ~ spec {
                        # (5 unchanged attributes hidden)
        
                      ~ containers {
                          + args        = []
                          + command     = []
                            name        = null
                            # (2 unchanged attributes hidden)
        
                          ~ resources {
                              + requests = {}
                                # (1 unchanged attribute hidden)
                            }
        
                            # (2 unchanged blocks hidden)
                        }
                    }
                }
        
                # (1 unchanged block hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccCloudRunService_cloudRunServiceGpuExample (25.27s)
FAIL

```


### Test Debug Log

- [link](https://storage.cloud.google.com/teamcity-logs/nightly/TerraformProviders_GoogleCloud_GOOGLE_BETA_NIGHTLYTESTS/2025-03-22/fc1d7e3/debug-google-beta-fc1d7e3-342398-TestAccCloudRunService_cloudRunServiceGpuExample.txt)

--- a/mmv1/templates/terraform/constants/cloud_run_service.go.tmpl
+++ b/mmv1/templates/terraform/constants/cloud_run_service.go.tmpl
@@ -9,6 +9,7 @@ func revisionNameCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, v i
 
 var cloudRunGoogleProvidedTemplateAnnotations = regexp.MustCompile(`template\.0\.metadata\.0\.annotations\.run\.googleapis\.com/sandbox`)
 var cloudRunGoogleProvidedTemplateAnnotations_autoscaling_maxscale = regexp.MustCompile(`template\.0\.metadata\.0\.annotations\.autoscaling\.knative\.dev/maxScale`)
+var cloudRunGoogleProvidedTemplateAnnotations_gpu_zonal_redundancy_disabled = regexp.MustCompile(`template\.0\.metadata\.0\.annotations\.run\.googleapis\.com/gpu-zonal-redundancy-disabled`)
 
 func cloudrunTemplateAnnotationDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 	// Suppress diffs for the annotations provided by API
@@ -19,6 +20,16 @@ func cloudrunTemplateAnnotationDiffSuppress(k, old, new string, d *schema.Resour
 
 	if cloudRunGoogleProvidedTemplateAnnotations_autoscaling_maxscale.MatchString(k) && new == "" {
 		return true
+	}
+
+	if cloudRunGoogleProvidedTemplateAnnotations_gpu_zonal_redundancy_disabled.MatchString(k) && new == "" {
+		if limitsRaw, ok := d.GetOk("template.0.spec.0.containers.0.resources.0.limits"); ok {
+			if limits, ok := limitsRaw.(map[string]interface{}); ok {
+				if _, hasGpu := limits["nvidia.com/gpu"]; hasGpu {
+					return true
+				}
+			}
+		}
 	}
 
 	// For other keys, don't suppress diff.

--- a/mmv1/templates/terraform/samples/services/cloudrun/cloud_run_service_gpu.tf.tmpl
+++ b/mmv1/templates/terraform/samples/services/cloudrun/cloud_run_service_gpu.tf.tmpl
@@ -2,17 +2,16 @@ resource "google_cloud_run_service" "{{$.PrimaryResourceId}}" {
   name     = "{{index $.ResourceIdVars "cloud_run_service_name"}}"
   location = "us-central1"
 
-  metadata {
-    annotations = {
-      "run.googleapis.com/launch-stage" = "BETA"
-    }
-  }
 
   template {
     metadata {
       annotations = {
         "autoscaling.knative.dev/maxScale": "1"
         "run.googleapis.com/cpu-throttling": "false"
+        # Explicitly disable zonal redundancy to bypass quota limits in the HashiCorp test project.
+        # Alternatively, if quota is granted to the test project, this annotation can be removed 
+        # to properly test the DiffSuppressFunc.
+        "run.googleapis.com/gpu-zonal-redundancy-disabled": "true"
       }
     }
     spec {

